src/test/java/teammates/test/cases/ui/browsertests/InstructorFeedbackPageUiTest.java

```diff
@@ -696,23 +699,25 @@ public void testDatePickerScripts() throws ParseException {
                      feedbackPage.getMinDateOf(Const.ParamsNames.FEEDBACK_SESSION_PUBLISHDATE));
         
         
-        ______TS("decreasing end date affects start time, visible time");
+        ______TS("decreasing end date does not affects start time or visible time");
+        initialCal.setTime(cal.getTime());
         cal.add(Calendar.DATE, -50);
         feedbackPage.fillTimeValueForDatePickerTest(Const.ParamsNames.FEEDBACK_SESSION_ENDDATE, cal);
         
-        assertEquals(sdf.format(cal.getTime()),
+        assertEquals(sdf.format(initialCal.getTime()),
                      feedbackPage.getValueOfDate(Const.ParamsNames.FEEDBACK_SESSION_STARTDATE));
         
-        assertEquals(sdf.format(cal.getTime()),
+        assertEquals(sdf.format(initialCal.getTime()),
                      feedbackPage.getValueOfDate(Const.ParamsNames.FEEDBACK_SESSION_VISIBLEDATE));
-        assertEquals(sdf.format(cal.getTime()),
+        assertEquals(sdf.format(initialCal.getTime()),
                      feedbackPage.getMaxDateOf(Const.ParamsNames.FEEDBACK_SESSION_VISIBLEDATE));
```

Comment:
One is checking for maxDate and one is checking for the actual value, I guess the code is little hard to read. I can introduce some variables to explain them instead